src/game/Objects/GameObject.cpp

```diff
@@ -1161,10 +1166,38 @@ void GameObject::TriggerLinkedGameObject(Unit* target)
     }
 
     // found correct GO
-    if (trapGO)
+    if (trapGO && trapGO->isSpawned())
         trapGO->Use(target);
 }
 
+void GameObject::RespawnLinkedGameObject()
+{
+    uint32 trapEntry = GetGOInfo()->GetLinkedGameObjectEntry();
+
+    if (!trapEntry)
+        return;
+
+    GameObjectInfo const* trapInfo = sGOStorage.LookupEntry<GameObjectInfo>(trapEntry);
+    if (!trapInfo || trapInfo->type != GAMEOBJECT_TYPE_TRAP)
+        return;
+
+    float range = 0.5f;
+
+    // search nearest linked GO
+    GameObject* trapGO = NULL;
+    {
+        // search closest with base of used GO, using max range of trap spell as search radius
```

Comment:
max range is 0.5f?

Reply:
Its the same range used when triggering it:
https://github.com/GuybrushGit/core/blob/f1330ef6c625d4d05ae63140f49f8c29dfef4be0/src/game/Objects/GameObject.cpp#L1149-L1152